composer.json

```diff
@@ -32,7 +32,7 @@
 		"ezyang/htmlpurifier": "^4.7",
 		"friendica/json-ld": "^1.0",
 		"league/html-to-markdown": "^4.8",
-		"level-2/dice": "^4",
+		"level-2/dice": "dev-master",
```

Comment:
Nooooo, what its the purpose of this drudgery?

Reply:
This one: https://github.com/Level-2/Dice/issues/179
I've an eye on it.. If there's a new release, I immediately will change the version back ..

Reply:
btw. this could maybe help us at `Factory` and `Repository` as well:
```php
use Dice;

class BaseRepository
{
        /** @var Dice */
        protected $dice;

         public function __construct(Dice $dice)
         {
                $this->dice = $dice;
         }
}
```

```php
use Friendica\Model\Introduction as ModelIntro;

class Introduction extends BaseRepository
{
         public function create()
         {
                $this->dice->create(ModelIntro::class);
         }
}
```

This avoids the `courier anti-pattern` see here: https://r.je/oop-courier-anti-pattern

Reply:
Isn't passing a `Dice` dependency a massive courier anti-pattern? Or a global container anti-pattern?

Reply:
Hmmmm fair enough... Let me rethink it again ^^

Reply:
I found another (my main) reason for it:
If there will be ever an storage addon, we'd have to extend this factory each time. So `$this->dice->create()` is working the same way.

Reply:
I'm afraid I'm not following.

Reply:
If we don't use Dice for automatic instancing, we would have to know which concrete class we're instancing, because we cannot be sure if the concrete `Storage` class won't extend the constructor with other parameters we would miss.

But maybe there's a more smart solution:
A new Hook, which is called in cases we couldn't create an instance, where the add-on is reponsible for creating an instance and add it. Like we do with notifies for the public server add-on :)